src/goto-programs/initialize_goto_model.cpp

```diff
@@ -133,10 +133,35 @@ goto_modelt initialize_goto_model(
   }
   else if(!binaries_provided_start)
   {
-    // Allow all language front-ends to try to provide the user-specified
-    // (--function) entry-point, or some language-specific default:
-    entry_point_generation_failed=
-      language_files.generate_support_functions(goto_model.symbol_table);
+    if(options.is_set("function"))
+    {
+      // no entry point is present; Use the mode of the specified entry function
+      // to generate one
+      auto const &entry_function_name = options.get_option("function");
+      auto const entry_function_sym =
+        goto_model.symbol_table.lookup({entry_function_name});
+      if(entry_function_sym == nullptr)
+      {
+        throw invalid_command_line_argument_exceptiont{
+          std::string{"couldn't find function with name `"} +
+            entry_function_name + "' in symbol table",
+          "--function"};
+      }
+      auto const entry_language =
+        get_language_from_mode(entry_function_sym->mode);
+      CHECK_RETURN(entry_language != nullptr);
```

Comment:
I guess one question is: `Should the entry point *always* have a mode?` - if so, then the assumption/assertion is the right thing to do. If there are valid cases where the mode might not be set, then it might be good to have some kind of diagnostic warning somewhere?

Reply:
I believe the mode should always be set. If that isn't the case right now, then we need to fix that.

Reply:
To make a potential failure easier to debug, it might be good to add a `PRECONDITION(!entry_function_sym->mode.empty());`